src/v2/blobs.rs

```diff
@@ -62,60 +47,27 @@ impl Client {
                 }
             }
         };
-        let req = match self.new_request(hyper::Method::GET, url.clone()) {
-            Ok(r) => r,
-            Err(e) => {
-                let msg = format!("new_request failed: {}", e);
-                error!("{}", msg);
-                return Box::new(futures::future::err::<_, _>(Error::from(msg)));
-            }
-        };
-        let freq = self.hclient.request(req);
-        let fres = freq
-            .from_err()
-            .inspect(move |_| {
-                trace!("GET {:?}", url);
-            }).and_then(move |r| {
-                match r.status() {
-                    StatusCode::MOVED_PERMANENTLY
-                    | StatusCode::TEMPORARY_REDIRECT
-                    | StatusCode::FOUND => {
-                        trace!("Got moved status {:?}", r.status());
-                    }
-                    _ => return Either::A(future::ok(r)),
-                };
-                let redirect: Option<String> = match r.headers().get("Location") {
-                    None => {
-                        return Either::A(future::err(Error::from(
-                            "get_blob: missing location header",
-                        )))
-                    }
-                    Some(loc) => {
-                        trace!("Got Location header {:?}", loc);
-                        String::from_utf8(loc.as_bytes().to_vec()).ok()
+
+        let fres = reqwest::async::Client::new()
+            .get(url)
+            .send()
+            .inspect(|res| trace!("Blob GET result: {:?}", res.status()))
+            .and_then(|mut res| {
+                let body = std::mem::replace(res.body_mut(), reqwest::async::Decoder::empty());
+                body.concat2()
+            }).map_err(|e| Error::from(format!("{}", e)))
+            .and_then(|body| {
+                let mut cursor = std::io::Cursor::new(body);
+                let mut buf = Vec::new();
+                use std::io::Read;
+                match cursor.read_to_end(&mut buf) {
```

Comment:
This is a blocking operation which would be better not to perform inside a future chain. I'm not sure what you are trying to do here, transforming a `Chunk` a into `Vec<u8>`?

Reply:
>  I'm not sure what you are trying to do here, transforming a `Chunk` into `Vec<u8>`

Yes.

> This is a blocking operation which would be better not to perform inside a future chain.
 
I'd be happy to receive some help with one then :-)